src/history/HistoryArchiveManager.cpp

```diff
@@ -156,6 +158,31 @@ HistoryArchiveManager::selectRandomReadableHistoryArchive() const
     }
 }
 
+void
+HistoryArchiveManager::logReportOnArchiveConfig() const
+{
+    auto& ws = mApp.getWorkScheduler();
+    for (auto const& archive : mArchives)
+    {
+        auto getHas = ws.executeWork<GetHistoryArchiveStateWork>(
```

Comment:
I don't think it's a good idea to have this endpoint synchronously process all archives: downloading multiple files can take a considerable amount of time and stall the node (this will have especially negative consequences for validating nodes). In the past we've seen `curl` taking abnormally long amounts of time, so I think we should make this reporting mechanism safer. Specifically, we should not be using `executeWork` and switch to `scheduleWork` (this will execute the work asynchronously). To manage  GetHistoryArchiveStateWork works, we could add a small class that derives from Work and checks on the status of its children.

Reply:
I'm not sure if I did it correctly, but I did try to create a class `historyArchiveReportWork` that derives from `Work` and checks on the status of its children each of which is a `GetHistoryArchiveStateWork`. Let me know what you think!